src/jobs/models/manager.ts

```diff
@@ -138,34 +161,129 @@ export class JobManager {
     await this.prisma.job.update(updateQueryBody);
   }
 
+  public async deleteJob(jobId: string): Promise<void> {
+    const deleteStages = this.prisma.stage.deleteMany({
+      where: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        job_id: jobId,
+      },
+    });
+
+    const deleteJob = this.prisma.job.delete({
+      where: {
+        id: jobId,
+      },
+    });
+
+    try {
+      await this.prisma.$transaction([deleteStages, deleteJob]);
```

Comment:
dont you want to define cascade delete instead? https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/referential-actions

Reply:
good idea, changed